mmdet/models/layers/transformer.py

```diff
@@ -401,659 +389,214 @@ def inverse_sigmoid(x, eps=1e-5):
+            query = self.post_norm(query)
+            if self.return_intermediate:
+                intermediate.pop()
+                intermediate.append(query)
 
-        Returns:
-            tuple[Tensor]: results of decoder containing the following tensor.
-
-                - out_dec: Output from decoder. If return_intermediate_dec \
-                      is True output has shape [num_dec_layers, bs,
-                      num_query, embed_dims], else has shape [1, bs, \
-                      num_query, embed_dims].
-                - memory: Output results from encoder, with shape \
-                      [bs, embed_dims, h, w].
-        """
-        bs, c, h, w = x.shape
-        # use `view` instead of `flatten` for dynamically exporting to ONNX
-        x = x.view(bs, c, -1).permute(2, 0, 1)  # [bs, c, h, w] -> [h*w, bs, c]
-        pos_embed = pos_embed.view(bs, c, -1).permute(2, 0, 1)
-        query_embed = query_embed.unsqueeze(1).repeat(
-            1, bs, 1)  # [num_query, dim] -> [num_query, bs, dim]
-        mask = mask.view(bs, -1)  # [bs, h, w] -> [bs, h*w]
-        memory = self.encoder(
-            query=x,
-            key=None,
-            value=None,
-            query_pos=pos_embed,
-            query_key_padding_mask=mask)
-        target = torch.zeros_like(query_embed)
-        # out_dec: [num_layers, num_query, bs, dim]
-        out_dec = self.decoder(
-            query=target,
-            key=memory,
-            value=memory,
-            key_pos=pos_embed,
-            query_pos=query_embed,
-            key_padding_mask=mask)
-        out_dec = out_dec.transpose(1, 2)
-        memory = memory.permute(1, 2, 0).reshape(bs, c, h, w)
-        return out_dec, memory
+        if self.return_intermediate:
+            return torch.stack(intermediate)
 
+        return query
 
-@MODELS.register_module()
-class DeformableDetrTransformerDecoder(TransformerLayerSequence):
-    """Implements the decoder in DETR transformer.
 
-    Args:
-        return_intermediate (bool): Whether to return intermediate outputs.
-        coder_norm_cfg (dict): Config of last normalization layer. Default：
-            `LN`.
-    """
+class DetrTransformerEncoderLayer(BaseModule):
 
-    def __init__(self, *args, return_intermediate=False, **kwargs):
+    def __init__(self,
+                 self_attn_cfg=dict(embed_dims=256, num_heads=8, dropout=0.0),
+                 ffn_cfg=dict(
+                     embed_dims=256,
+                     feedforward_channels=1024,
+                     num_fcs=2,
+                     ffn_drop=0.,
+                     act_cfg=dict(type='ReLU', inplace=True)),
+                 norm_cfg=dict(type='LN'),
+                 init_cfg=None,
+                 batch_first=False):
 
-        super(DeformableDetrTransformerDecoder, self).__init__(*args, **kwargs)
-        self.return_intermediate = return_intermediate
+        super(DetrTransformerEncoderLayer, self).__init__(init_cfg)
```

Comment:
`super(DetrTransformerEncoderLayer, self).__init__(init_cfg)` -> `super().__init__(init_cfg=init_cfg)`

Reply:
OK, thanks for your suggestion !